drivers/timer/hpet.c

```diff
@@ -378,15 +390,15 @@ void sys_clock_set_timeout(int32_t ticks, bool idle)
 		return;
 	}
 
-	ticks = ticks == K_TICKS_FOREVER ? max_ticks : ticks;
-	ticks = CLAMP(ticks - 1, 0, (int32_t)max_ticks);
+	ticks = ticks == K_TICKS_FOREVER ? HPET_MAX_TICKS : ticks;
```

Comment:
`MAX_TICKS` and `max_ticks` are both calculated value based on cycles per tick. And you are simply replacing them with `0x7FFFFFFF`. These are used so that we get at least one timer interrupt before the clock counter overflows. Simply replacing them with a constant is not right.

Reply:
before HPET works in 32-bit mode, so its MAX_TICK must be calculated at runtime.
now it's switched to 64-bit mode, 0x7fffffff is for sure supported.